third_party/move/tools/move-unit-test/src/lib.rs

```diff
@@ -218,8 +228,31 @@ impl UnitTestingConfig {
         genesis_state: Option<ChangeSet>,
         cost_table: Option<CostTable>,
         writer: W,
+    ) -> Result<(W, bool)> {
+        self.run_and_report_unit_tests_with_gas_meter(
+            test_plan,
+            native_function_table,
+            genesis_state,
+            writer,
+            GasStatus::new(
+                &cost_table.unwrap_or_else(unit_cost_table),
+                Gas::new(self.gas_limit.unwrap_or(DEFAULT_EXECUTION_BOUND)),
+            ),
```

Comment:
Note: `GasStatus` is outdated legacy code. We really should get rid of it.

If you need to count the number of instructions I'd recommend you implement your a custom gas meter that does this.